src/ripple/protocol/SOTemplate.h

```diff
@@ -61,50 +75,44 @@ class SOElement
 class SOTemplate
 {
 public:
-    using list_type = std::vector <std::unique_ptr <SOElement const>>;
-    using iterator_range = boost::iterator_range<list_type::const_iterator>;
+    SOTemplate(SOTemplate&& other) = default;
 
-    /** Create an empty template.
-        After creating the template, call @ref push_back with the
-        desired fields.
-        @see push_back
+    /** Create a template populated with all fields.
+        After creating the template fields cannot be
+        added, modified, or removed.
     */
-    SOTemplate () = default;
+    SOTemplate (std::initializer_list<SOElement> uniqueFields,
```

Comment:
I like your idea of using `std::initializer_list` much better than my approach (passing a lambda). I also like that we have the "common fields" separate from the "unique fields" because it helps avoid mistakes, but I'm not sure I'm sold on passing _two_ initializer lists.

Rationale: it pollutes the interface of `SOTemplate` with a particular implementation detail (namely that some fields are "common" to multiple templates and some fields are not).

It would sit better with me, I think, if the callers handled the "detail" of inserting the common fields alongside the unique fields and passing one thing to this constructor.